### PR TITLE
feat: Add ECDSA secp256k1 builtin test

### PR DESCRIPTION
**Pull request by kevaundray**
# Related issue(s)

Resolves # 

# Description

This adds back the ECDSA secp256k1 test, to see what the status of it is since UltraPlonk
## Test additions / changes

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.

## Documentation needs
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

**Comment by Globallager** [comment]
Closes https://github.com/noir-lang/noir/issues/1106 and closes https://github.com/noir-lang/noir/issues/12

**Comment by TomAFrench** [comment]
It could be helpful to move an equivalent of this across to [aztec_backend](https://github.com/noir-lang/aztec_backend/blob/c9fb9e806f1400a2ff7594a0669bec56025220bb/src/composer.rs#L639) as ideally we'd have a test for each black box function over there.

**Comment by kevaundray** [comment]
This now fails with a constraint not satisfied error which is perhaps because of a misconfiguration in acvm/pwg.

Before it was crashing -- Note, we are currently committed to a branch of barretenberg, so this is not mergable until that branch gets merged into bb/master

File: crates/nargo_cli/tests/prove_and_verify.rs [hunk 1]
```diff
@@ -72,6 +72,9 @@ mod tests {
                 test_dir.file_name().into_string().expect("Directory can't be converted to string");
             let test_program_dir = &test_dir.path();
 
+            if test_name != "ecdsa_secp256k1" {
```
Comment (kevaundray) [inline crates/nargo_cli/tests/prove_and_verify.rs:75]: Need to remove this before merging

File: flake.lock [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
       },
       "locked": {
-        "lastModified": 1683314474,
-        "narHash": "sha256-gfHYpOnVTfS+4fhScBhfkB/e5z+jPFCi8zSy+aEh+8s=",
+        "lastModified": 1683743846,
+        "narHash": "sha256-YEy3Qf94HDVBxc+wztsuneeL7xLlcJ+0LxOjIyfPvhY=",
         "owner": "AztecProtocol",
         "repo": "barretenberg",
-        "rev": "ad615ee7dc931d3dbea041e47c96b9d8dccebf98",
+        "rev": "964f22a83571442fa959869e72070bf5c4330995",
```
Comment (kevaundray) [inline flake.lock:17]: Need to change to an upstream master version of barretenberg before merging

File: crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+     assert(hashed_message == expected);
+
+     let x = std::ecdsa_secp256k1::verify_signature(pub_key_x, pub_key_y, signature, hashed_message);
+     assert(x == 1);
+}
+
+// TODO: The following will fail in ssa: 
+//
+//
+//fn main(message : [u8;38],hashed_message : [u8;32], _pub_key_x : [u8;32], _pub_key_y : [u8;32], _signature : [u8;64]) {
```
Comment (kevaundray) [inline crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr:25]: Need to create an issue before merging

Comment (kevaundray) [reply crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr]: Added a regression PR to show this failing currently

**Comment by kevaundray** [comment]
Hmm tests taking so long implies that the nix cache may not be working fully

File: crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr [hunk 4]
```diff
@@ -0,0 +1,11 @@
+use dep::std;
+
+
+fn main(message : [u8;38],hashed_message : [u8;32], pub_key_x : [u8;32], pub_key_y : [u8;32], signature : [u8;64]) {
+     // Hash the message, since secp256k1 expects a hashed_message
+     let expected= std::hash::sha256(message);
```
Comment (vezenovm) [inline crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr:6]: extreme nit, space after "let expected"

File: crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+     assert(hashed_message == expected);
+
+     let x = std::ecdsa_secp256k1::verify_signature(pub_key_x, pub_key_y, signature, hashed_message);
```
Comment (vezenovm) [inline crates/nargo_cli/tests/test_data/ecdsa_secp256k1/src/main.nr:9]: nit: wouldn't a better name be is_verified or verified?

**Review by vezenovm — APPROVED** [review]
Just some nits, good to merge with tests passing

**Pull request merged**